packages/framework-provider-azure-infrastructure/src/infrastructure/synth/terraform-service-plan.ts

```diff
@@ -0,0 +1,24 @@
+import { TerraformStack } from 'cdktf'
+import { servicePlan, resourceGroup } from '@cdktf/provider-azurerm'
+import { toTerraformName } from '../helper/utils'
+import { AzurermProvider } from '@cdktf/provider-azurerm/lib/provider'
+
+export class TerraformServicePlan {
+  static build(
+    providerResource: AzurermProvider,
+    terraformStackResource: TerraformStack,
+    group: resourceGroup.ResourceGroup,
+    appPrefix: string,
+    resourceGroupName: string
+  ): servicePlan.ServicePlan {
+    const id = toTerraformName(appPrefix, 'hpn')
+    return new servicePlan.ServicePlan(terraformStackResource, id, {
+      name: `${resourceGroupName}hpn`,
+      location: group.location,
+      resourceGroupName: group.name,
+      osType: 'Windows',
+      skuName: 'Y1',
```

Comment:
Curiosity: What this magic string refers to?

Reply:
For Function Apps the possible values are: Elastic and Consumption SKUs (Y1, EP1, EP2, and EP3)

The `Y1` means:
  * Tear: Dynamic
  * Full name: A  function consumption plan

Reply:
Aha, thanks